index.js

```diff
@@ -38,16 +38,16 @@ util.inherits(AutoDetectDecoderStream, Transform);
  * @param {function} done
  */
 AutoDetectDecoderStream.prototype._consumeBufferForDetection = function (chunk, done) {
+    if (!this._detectionBuffer) {
```

Comment:
Have you come across a situation where there is no `chunk` here?!

Reply:
Well you do send a null chunk on line 124

Reply:
Yea exactly!